### PR TITLE
Prevent inventory events from being relayed to pockets

### PR DESCRIPTION
**Pull request by ElectroJr**
Currently shoving insulated gloves or galoshes in your pockets works just as well as wearing them. This PR fixes this by just excluding pockets from the inventory event relay.

:cl:
- fix: Insulated gloves and galoshes no longer work when in your pockets. You actually have to wear them.

### CODE REVIEW

File: Content.Shared/Inventory/InventorySystem.Relay.cs [hunk 1]
```diff
@@ -18,7 +18,10 @@ public void InitializeRelay()
 
     protected void RelayInventoryEvent<T>(EntityUid uid, InventoryComponent component, T args) where T : EntityEventArgs
     {
-        var containerEnumerator = new ContainerSlotEnumerator(uid, component.TemplateId, _prototypeManager, this);
+        // Insulated gloves in you pockets ain't gonna save you.
+        var flags = ~SlotFlags.POCKET; 
+
+        var containerEnumerator = new ContainerSlotEnumerator(uid, component.TemplateId, _prototypeManager, this, flags);
```
Comment (moonheart08) [inline Content.Shared/Inventory/InventorySystem.Relay.cs:24]: Would this not make more sense as some sort of opt-out? I can imagine scenarios where you'd want to relay events to in-pocket items.

**Review by moonheart08 — CHANGES_REQUESTED** [review]

File: Content.Shared/Inventory/InventorySystem.Relay.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Relay.cs:24]: Well you could opt out by just not using the generic `RelayInventoryEvent<T>()` subscription. But if there should be a generic way of having each event specify its own target slots, then I think you'd have to do it by adding some sort of `IInventoryRelayEvent` interface that those events implement which requires them to specify target slots? Unless someone can think of something better I can add that.

Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Relay.cs]: I've now just added an `IInventoryRelayEvent`

**Review by moonheart08 — APPROVED** [review]

File: Content.Shared/Inventory/InventorySystem.Slots.cs [hunk 2]
```diff
@@ -109,9 +111,18 @@ public bool MoveNext([NotNullWhen(true)] out ContainerSlot? container)
             container = null;
             if (_nextIdx >= _slots.Length) return false;
 
-            while (_nextIdx < _slots.Length && !_inventorySystem.TryGetSlotContainer(_uid, _slots[_nextIdx++].Name, out container, out _)) { }
+            while (_nextIdx < _slots.Length)
+            {
+                var slot = _slots[_nextIdx++];
```
Comment (ShadowCommander) [inline Content.Shared/Inventory/InventorySystem.Slots.cs:116]:
```suggestion
            for (; _nextIdx < _slots.Length; _nextIdx++)
            {
                var slot = _slots[_nextIdx];
```

**Review by ShadowCommander — CHANGES_REQUESTED** [review]
Other than that looks good.

**Review by ShadowCommander — APPROVED** [review]

**Review by moonheart08 — CHANGES_REQUESTED** [review]
The tests have hung. Seems you have an infinite loop somewhere?

File: Content.Shared/Inventory/InventorySystem.Slots.cs (hunk 2 above)
Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Slots.cs]: enumerator returns in the middle of the loop, so the increment from the for loop never happens.
reverting to a while loop with explicit increment.

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**